src/jit/morph.cpp

```diff
@@ -2711,13 +2626,15 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
     const unsigned maxRegArgs = MAX_REG_ARG; // other arch: fixed constant number
 #endif
 
-    unsigned argSlots                = 0;
-    unsigned nonRegPassedStructSlots = 0;
-    bool     reMorphing              = call->AreArgsComplete();
-    bool     callHasRetBuffArg       = call->HasRetBufArg();
-    bool     callIsVararg            = call->IsVarargs();
+    if (call->fgArgInfo != nullptr)
+    {
+        // We've already initialized and set the fgArgInfo.
```

Comment:
Should this be an error (assert)? why do we allow it?

Reply:
Currently the tail call transformation has to deduce all the information that is initialized here. We want this to be callable by that transformation, and then it will be initialized and won't have to be recomputed.